### DISCUSSION TITLE
No class based bot

### DISCUSSION THREAD

**Discussion by anselal** [Ext-Creators/discord-ext-ipc#37] [Q&A]
Hi there,

I am using a none class based bot approach and want to ask what the decorator should be for `on_ipc_ready` and `on_ipc_error`.

Is it `@bot.event` ???

**Comment by lgaan** [comment]
The examples are incorrect, you can’t define custom events within a subclass unless you manually call `self.add_listener(self.on_ipc_ready, "on_ipc_ready")` etc.

Otherwise you’ll need to use `@bot.event` and register the events outside of the subclass.

The examples will be updated soon to reflect this.

**Comment by anselal** [comment]
But the message inside `on_ipc_ready` is printed which means the event is called.

**Comment by anselal** [comment]
Can you let me know when the examples get updated ?
